erts/emulator/test/match_spec_SUITE.erl

```diff
@@ -239,6 +240,85 @@ test_5b(Config) when is_list(Config) ->
+    erlang:trace(P, true, [call]),
+    %% Notice `local` function tracing
+    erlang:trace_pattern({?MODULE, f2_test6, 1}, Pat, [local]),
+    erlang:trace_pattern({?MODULE, f1_test6, 0}, Pat, [local]),
+    collect(P, [{trace, P, call, {?MODULE, f2_test6, [f1]},
+                   [
+                    {?MODULE, f3_test6, 0, [{file, "test6.erl"}, {line, 21}]},
+                    {?MODULE, f5_test6, 0, [{file, "test6.erl"}, {line, 14}]},
+                    {?MODULE, fixed_runner, 2, [{file, "test6.erl"}, {line, 7}]}
+                   ]},
+                {trace, P, call, {?MODULE, f1_test6, []},
+                   [
+                    {?MODULE, f2_test6, 1, [{file, "test6.erl"}, {line, 25}]},
+                    {?MODULE, f3_test6, 0, [{file, "test6.erl"}, {line, 21}]},
+                    {?MODULE, f5_test6, 0, [{file, "test6.erl"}, {line, 14}]},
+                    {?MODULE, fixed_runner, 2, [{file, "test6.erl"}, {line, 7}]}
+                   ]}
+               ]),
+
+    Pat2 = [{'_', [], [{message, {current_stacktrace, 3}}]}],
+    P2 = spawn(?MODULE, fixed_runner, [self(), Fun]),
+    erlang:trace(P2, true, [call]),
+    %% Notice `local` function tracing
+    erlang:trace_pattern({?MODULE, f2_test6, 1}, Pat2, [local]),
+    erlang:trace_pattern({?MODULE, f1_test6, 0}, Pat2, [local]),
+    collect(P2, [{trace, P2, call, {?MODULE, f2_test6, [f1]},
+                   [
+                    {?MODULE, f3_test6, 0, [{file, "test6.erl"}, {line, 21}]},
+                    {?MODULE, f5_test6, 0, [{file, "test6.erl"}, {line, 14}]},
+                    {?MODULE, fixed_runner, 2, [{file, "test6.erl"}, {line, 7}]}
+                   ]},
+                {trace, P2, call, {?MODULE, f1_test6, []},
+                   [
+                    {?MODULE, f2_test6, 1, [{file, "test6.erl"}, {line, 25}]},
+                    {?MODULE, f3_test6, 0, [{file, "test6.erl"}, {line, 21}]},
+                    {?MODULE, f5_test6, 0, [{file, "test6.erl"}, {line, 14}]}
+                   ]}
+               ]),
+
+    %% Test when erts_backtrace_depth is less than depth
+    erlang:system_flag(backtrace_depth, 2),
+    P3 = spawn(?MODULE, fixed_runner, [self(), Fun]),
+    erlang:trace(P3, true, [call]),
+    %% Notice `local` function tracing
+    erlang:trace_pattern({?MODULE, f2_test6, 1}, Pat2, [local]),
+    erlang:trace_pattern({?MODULE, f1_test6, 0}, Pat2, [local]),
+    collect(P3, [{trace, P3, call, {?MODULE, f2_test6, [f1]},
+                   [
+                    {?MODULE, f3_test6, 0, [{file, "test6.erl"}, {line, 21}]},
+                    {?MODULE, f5_test6, 0, [{file, "test6.erl"}, {line, 14}]}
+                   ]},
+                {trace, P3, call, {?MODULE, f1_test6, []},
+                   [
+                    {?MODULE, f2_test6, 1, [{file, "test6.erl"}, {line, 25}]},
+                    {?MODULE, f3_test6, 0, [{file, "test6.erl"}, {line, 21}]}
+                   ]}
+               ]),
```

Comment:
We had a bunch of failed testcases on account of `backtrace_depth` not being restored to what it was before. Try restoring it when you're done, e.g.

```suggestion
    %% Test when erts_backtrace_depth is less than depth
    OldDepth = erlang:system_flag(backtrace_depth, 2),
    try
        P3 = spawn(?MODULE, fixed_runner, [self(), Fun]),
        erlang:trace(P3, true, [call]),
        %% Notice `local` function tracing
        erlang:trace_pattern({?MODULE, f2_test6, 1}, Pat2, [local]),
        erlang:trace_pattern({?MODULE, f1_test6, 0}, Pat2, [local]),
        collect(P3, [{trace, P3, call, {?MODULE, f2_test6, [f1]},
                      [
                       {?MODULE, f3_test6, 0, [{file, "test6.erl"}, {line, 21}]},
                       {?MODULE, f5_test6, 0, [{file, "test6.erl"}, {line, 14}]}
                      ]},
                     {trace, P3, call, {?MODULE, f1_test6, []},
                      [
                       {?MODULE, f2_test6, 1, [{file, "test6.erl"}, {line, 25}]},
                       {?MODULE, f3_test6, 0, [{file, "test6.erl"}, {line, 21}]}
                      ]}
                    ])
    after
        erlang:system_flag(backtrace_depth, OldDepth)
    end,
```

Reply:
Oh! I apologize for the trouble. I will apply this.

But first I want to consult you about something. I remembered the difference of clamping the depth during execution instead of compilation. When setting the backtrace_depth flag, the stack tracing will immediately update instead of changing only when compiled again. Which is the correct behavior?

Reply:
Setting it on compilation is the correct behavior IMO.